tutor/src/components/task-step/exercise.cjsx

```diff
@@ -123,27 +123,21 @@ module.exports = React.createClass
     @setCurrentStep(current)
 
   setCurrentStep: (currentStep) ->
-    return unless currentStep isnt @state.currentStep
-
+    return if currentStep is @state.currentStep
     @setState({currentStep})
     @props.goToStep(currentStep)
 
-  setCurrentStepByStepId: (id) ->
-    {taskId} = @props
-    stepNavIndex = TaskStore.getStepNavIndex(taskId, id)
-    @setCurrentStep(stepNavIndex)
-
-  onFreeResponseChange: (id, tempFreeResponse) ->
+  onFreeResponseChange: (id, tempFreeResponse, stepIndex) ->
     TaskStepActions.updateTempFreeResponse(id, tempFreeResponse)
 
     # set part to be active if part of multipart
-    @setCurrentStepByStepId(id) unless @isSinglePart(@state.parts)
+    @setCurrentStep(stepIndex) unless @isSinglePart(@state.parts)
```

Comment:
Sorry, I didn't make this very clear with comments, but this `stepIndex` getting passed in is the step index based on the actual existing steps.  However, since we've gotten more complicated with our interstitial cards stuffs, this step index does not match the index of the steps on the front end when there are interstitial introduction cards, hence the ugly `TaskStore.getStepNavIndex` stuffs, because that's what `setCurrentStep` wants -- the index that accounts for all crumbs/interstitial stuffs.  I believe these changes will cause weird behavior when the MPQ is after any interstitial cards.  Potentially the current bug is due to unsynced "stepIndices" as well, which the current steps cleanup stuffs in #1354 should help with.